### PR TITLE
test: usePrevious hook

### PR DESCRIPTION
**Pull request by yardz**
Tests for `usePrevious` hook.

### CODE REVIEW

File: superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { renderHook } from '@testing-library/react-hooks';
+import { usePrevious } from './usePrevious';
+
+test('get undefined on frist render when initalValue is not defined', () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts:22]: get undefined on the first render when initialValue is not defined

File: superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const hook = renderHook(() => usePrevious('state'));
+  expect(hook.result.current).toBeUndefined();
+});
+
+test('get initail value on frist render when initalValue is defined', () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts:27]: get initial value on the first render when initialValue is defined

File: superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts (hunk 1 above)
Comment (yardz) [reply superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts]: Tnks

File: superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts (hunk 2 above)
Comment (yardz) [reply superset-frontend/src/common/hooks/usePrevious/usePrevious.test.ts]: tnks 2 rsrs

**Comment by yardz** [comment]
@nytai or @ktmud Can either of you see this PR for me? Thanks!

**Review by ktmud — APPROVED** [review]
This is nice and thanks for adding these tests.

Before we add more utility hooks and spend efforts in adding tests, I'm wondering whether we can utilize some existing libraries? For example, this `usePrevious` hook is available in all of [react-use](https://github.com/streamich/react-use/blob/master/docs/usePrevious.md), [ahooks](https://ahooks.js.org/hooks/state/use-previous) (5k GitHub stars) and [react-recipes](https://github.com/craig1123/react-recipes/blob/master/docs/usePrevious.md) (447 stars), and they all have their own community-contributed unittests.

cc @nytai @rusackas @villebro @suddjian

**Comment by suddjian** [comment]
I have nothing against those libraries, but in my opinion we don't currently need a new dependency for simple utilities like this. 

- The vast majority of hooks from those libraries are not needed right now.
- Dependencies need to be evaluated, upgraded, add to build times (`react-use` is already on major version 17!)
- Many people are still getting used to hooks, and keeping simple hooks like this in our codebase will help them to learn.
- Hooks are a new enough pattern that the "best choice" of utility library is likely to change.

**Review by michael-s-molina — APPROVED** [review]

**Pull request merged**